astro.config.mjs

```diff
@@ -1,9 +1,15 @@
 import { defineConfig } from "astro/config";
 import starlight from "@astrojs/starlight";
 import tailwind from "@astrojs/tailwind";
+import node from '@astrojs/node'; // Node.js adapter for SSR
 
 // https://astro.build/config
 export default defineConfig({
+  output: 'server', // Enables SSR
```

Comment:
This enables SSR for the **entire** site, not just an individual page that needs to be rerendered each time the page is viewed. I would optimally prefer a solution that puts the correct version in the HTML of the served page as per the most recent update, but then checks the schedule and updates it dynamically on-page if time has passed and a different lecture is the next upcoming one.

Reply:
Enabling SSR for the entire site ensures that every page is dynamically rendered with up-to-date content, including schedule changes. However, I can explore implementing client-side JavaScript to update the upcoming lecture dynamically on the page without requiring a full refresh if that's the preferred approach.

Reply:
are you just pasting my comments into chatgpt and asking it to write a response for you? I'm not going to respond to any more comments here.

Reply:
chris not exactly like that my english is not that good and i dont want ppl to get miscommunicated with my english thats why i did that if u want explain each line of code